logstash-core/lib/logstash/environment.rb

```diff
@@ -1,7 +1,40 @@
 # encoding: utf-8
 require "logstash/errors"
+require "logstash/config/cpu_core_strategy"
+require "logstash/settings"
 
 module LogStash
+
+  [
+            Setting::String.new("node.name", Socket.gethostname),
```

Comment:
If this style is undesirable, we could do this the same way Plugin config settings are given:

```ruby
setting "node.name", Setting::String, :default => Socket.gethostname
```

```ruby
def self.setting(name, type, **stuff)
  SETTINGS.register(klass.new(name, ...whtaever..))
end
```